"Module not found" when using --watch flag

I'm using v2.8.12 and am lazy loading some modules. When I save changes to one of those modules  in `--watch` mode and do a hard refresh, the error "Module not found" is thrown when loading it. I'm also using Babel loader.

```
13.077da2d9698e29cebe18.js:18565 Error: Module not found: core-js/modules/es.array.index-of
```

This issue doesn't happen when doing a regular build.

--- a/lib/DojoAMDModuleFactoryPlugin.js
+++ b/lib/DojoAMDModuleFactoryPlugin.js
@@ -356,6 +356,10 @@ module.exports = class DojoAMDModuleFactoryPlugin {
 	 * may have an absMid we can use for the context).
 	 */
 	buildModule(module) {
+		// Need to reset this for watch mode so require calls prior to the define function
+		// will be treated as cjs require when a module is re-compiled.
+		delete module.isAMD;
+
 		if (module.absMid || !module.rawRequest) return;
 		const dojoRequire = callSyncBail(this.compiler, "get dojo require");
 		const issuerAbsMid = module.issuer && module.issuer.absMid;

